test/Microsoft.AspNetCore.SignalR.Client.Tests/HttpConnectionTests.Negotiate.cs

```diff
@@ -66,12 +67,15 @@ public Task ConnectionCannotBeStartedIfNoCommonTransportsBetweenClientAndServer(
                         ResponseUtils.CreateNegotiationContent());
                 });
 
-                await WithConnectionAsync(
-                    CreateConnection(testHttpHandler, url: requestedUrl),
-                    async (connection) =>
-                    {
-                        await connection.StartAsync(TransferFormat.Text).OrTimeout();
-                    });
+                using (var noErrorScope = new VerifyNoErrorsScope())
```

Comment:
You could do something super evil and async localish so you don't have to pass the logger factory around 😄

Reply:
The main issue is passing it to whatever piece of code you're testing. Hard to do that automatically. It really invokes updating each test.

Reply:
Maybe it does. @JunTaoLuo just wrote an xunit test framework that we're going to start using for logging very soon. It does this sorta thing at a lower level to avoid updating all tests. I wonder if we can extract out a system that lets us basically wrap any test with begin end code.

Reply:
> basically wrap any test with begin end code.

Xunit has that... it's a constructor (begin) and Dispose method (end). https://xunit.github.io/docs/shared-context#constructor

Reply:
This could helper could move to aspnet/Logging or aspnet/Testing though :) (can wait until after this PR, during 2.2.0-mq or something)